lib/tarball.js

```diff
@@ -0,0 +1,58 @@
+const { relative } = require('path')
+
+const pkgContents = require('@npmcli/installed-package-contents')
+const pacote = require('pacote')
+const isPackageBin = require('pacote/lib/util/is-package-bin.js')
+const tar = require('tar')
+
+// applies same opts as pacote.tarball for directory npa specs
+const tarOpts = manifest => ({
+  cwd: manifest._resolved,
+  prefix: 'package/',
+  portable: true,
+  gzip: true,
+
+  // ensure that package bins are always executable
+  // Note that npm-packlist is already filtering out
+  // anything that is not a regular file, ignored by
+  // .npmignore or package.json "files", etc.
+  filter: (path, stat) => {
+    if (isPackageBin(manifest, path))
+      stat.mode |= 0o111
+    return true
+  },
+
+  // Provide a specific date in the 1980s for the benefit of zip,
+  // which is confounded by files dated at the Unix epoch 0.
+  mtime: new Date('1985-10-26T08:15:00.000Z'),
+})
+
+// returns a simplified tarball when reading files from node_modules folder,
+// thus avoiding running the prepare scripts and the extra logic from packlist
+const nodeModulesTarball = (manifest, opts) =>
+  pkgContents({ path: manifest._resolved, depth: 1 })
+    .then(files =>
+      files.map(file => relative(manifest._resolved, file))
+    )
+    .then(files =>
+      tar.c(tarOpts(manifest), files)
+        .on('error', /* istanbul ignore next */ err => {
+          throw err
```

Comment:
This won't have the effect it looks like.  Since the throw happens after the current tick, it'll escape the Promise chain.

I think this bit could be replaced with just `.then(files => tar.c(tarOpts(manifest), files).concat())`, since we're just calling `.concat()` on the stream, and `.concat()` already returns a Promise that rejects if the stream emits `error`.